### PR TITLE
[IMP] open_academy : relations between models #59945

### PR DESCRIPTION
**Pull request by JJRR5**
**FInished relations between models chapter** 
Created the model _sessions_ which main goal is to be able to create a session and this session could be related to a course, to do that I used the relations between models fields (many2one, one2many, many2many).

### CODE REVIEW

File: open_academy/__init__.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # -*- coding: utf-8 -*-
 
 from . import controllers
-from . import models
\ No newline at end of file
+from . import models
+
```
Comment (sebasdrk17) [inline open_academy/__init__.py:5]: just leave 1 blank space at the bottom of the file

File: open_academy/models/__init__.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
 # -*- coding: utf-8 -*-
 
-from . import models
\ No newline at end of file
+# from . import models
```
Comment (sebasdrk17) [inline open_academy/models/__init__.py:3]: if not needed, remove the comment

File: open_academy/models/course.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    _title = 'open_academy.course'
+    _description = 'open_academy.course'
+    
+    name = fields.Char()
+    title = fields.Text()
+    description = fields.Text()
+    responsible_user_id = fields.Many2one('res.users')
+    session_id = fields.One2many('open_academy.sessions','courses_id')   
+
+
```
Comment (sebasdrk17) [inline open_academy/models/course.py:15]: leave just one blank space at the bottom

File: open_academy/models/sessions.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    
+    name = fields.Char()
+    start_date = fields.Date(required=True)
+    duration = fields.Integer()
+    number_of_seats = fields.Integer()
+    instructor_id = fields.Many2one('res.partner')
+    courses_id = fields.Many2one('open_academy.course') 
+    attendees_ids = fields.Many2many("res.partner",string="Attendees")
+    stash = 1
+
```
Comment (sebasdrk17) [inline open_academy/models/sessions.py:16]: same as other, remove one space

File: open_academy/models/sessions.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    _description = 'This is a sessions model'
+    
+    name = fields.Char()
+    start_date = fields.Date(required=True)
+    duration = fields.Integer()
+    number_of_seats = fields.Integer()
+    instructor_id = fields.Many2one('res.partner')
+    courses_id = fields.Many2one('open_academy.course') 
+    attendees_ids = fields.Many2many("res.partner",string="Attendees")
+    stash = 1
```
Comment (sebasdrk17) [inline open_academy/models/sessions.py:15]: what's this for?

File: open_academy/models/sessions.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    _name = 'open_academy.sessions'
+    _description = 'This is a sessions model'
+    
+    name = fields.Char()
+    start_date = fields.Date(required=True)
+    duration = fields.Integer()
+    number_of_seats = fields.Integer()
+    instructor_id = fields.Many2one('res.partner')
+    courses_id = fields.Many2one('open_academy.course') 
+    attendees_ids = fields.Many2many("res.partner",string="Attendees")
```
Comment (sebasdrk17) [inline open_academy/models/sessions.py:14]:
```suggestion
    attendees_ids = fields.Many2many("res.partner", string="Attendees")
```

File: open_academy/models/sessions.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from odoo import models,fields
```
Comment (sebasdrk17) [inline open_academy/models/sessions.py:1]:
```suggestion
from odoo import models, fields
```

File: open_academy/models/course.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
+from odoo import models,fields
```
Comment (sebasdrk17) [inline open_academy/models/course.py:1]:
```suggestion
from odoo import models, fields
```

File: open_academy/views/form.xml [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<odoo>
+    <record  model="ir.ui.view" id="view_course_form">
```
Comment (sebasdrk17) [inline open_academy/views/form.xml:3]: swap positions between model and id

File: open_academy/views/search.xml [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        </field>
+    </record>
+</odoo>
+
+
+
+
+
+
+
```
Comment (sebasdrk17) [inline open_academy/views/search.xml:21]: Lots of white spaces

File: open_academy/views/view_course.xml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+<odoo>
+    <record model="ir.actions.act_window" id="action_menu_courses">
```
Comment (sebasdrk17) [inline open_academy/views/view_course.xml:2]: Swap positions between model and id

File: open_academy/views/view_sessions.xml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<odoo>
+    <record model="ir.actions.act_window" id="action_menu_sessions">
```
Comment (sebasdrk17) [inline open_academy/views/view_sessions.xml:2]: swap positions also

**Review by sebasdrk17 — COMMENTED** [review]
Once changes are done tag me again to review

File: open_academy/models/sessions.py (hunk 5 above)
Comment (JJRR5) [reply open_academy/models/sessions.py:15]: the stash variable was just for a test with the stash git command, I will delete it

**Comment by sebasdrk17** [comment]
Closing this as a new branch will be created with rebased code from stable.

**Pull request closed without merging**